src/lib.rs

```diff
@@ -170,6 +170,11 @@ pub fn download_firmware_id(firmware_id: &str) -> Result<(String, String), Strin
         Some(config::CERT)
     )?;
 
+    if !fs::metadata(config::CACHE).is_ok() {
```

Comment:
Is this just checking the directory exists? Then `!Path::new(config::CACHE).is_dir()` would be more direct.

Reply:
Yes it is, and good to know. Switched to use that one in 0244cfe2535780b6a75208a1c5e7b74579c53a9b.